pkg/porter/parameters.go

```diff
@@ -0,0 +1,306 @@
+	pset.Modified = pset.Created
+
+	if opts.DryRun {
+		data, err := json.Marshal(pset)
+		if err != nil {
+			return errors.Wrap(err, "unable to generate parameters JSON")
+		}
+		fmt.Fprintf(p.Out, "%v", string(data))
+		return nil
+	}
+	err = p.Parameters.Save(*pset)
+	return errors.Wrapf(err, "unable to save parameters")
+}
+
+// Validate validates the args provided to Porter's parameter show command
+func (o *ParameterShowOptions) Validate(args []string) error {
+	if err := validateParameterName(args); err != nil {
+		return err
+	}
+	o.Name = args[0]
+	return o.ParseFormat()
+}
+
+// Validate validates the args provided to Porter's parameter edit command
+func (o *ParameterEditOptions) Validate(args []string) error {
+	if err := validateParameterName(args); err != nil {
+		return err
+	}
+	o.Name = args[0]
+	return nil
+}
+
+// EditParameter edits the parameters of the provided name.
+func (p *Porter) EditParameter(opts ParameterEditOptions) error {
+	paramSet, err := p.Parameters.Read(opts.Name)
+	if err != nil {
+		return err
+	}
+
+	contents, err := yaml.Marshal(paramSet)
+	if err != nil {
+		return errors.Wrap(err, "unable to load parameters")
+	}
+
+	editor := editor.New(p.Context, fmt.Sprintf("porter-%s.yaml", paramSet.Name), contents)
+	output, err := editor.Run()
+	if err != nil {
+		return errors.Wrap(err, "unable to open editor to edit parameters")
+	}
+
+	err = yaml.Unmarshal(output, &paramSet)
+	if err != nil {
+		return errors.Wrap(err, "unable to process parameters")
+	}
+
+	err = p.Parameters.Validate(paramSet)
+	if err != nil {
+		return errors.Wrap(err, "parameters are invalid")
+	}
+
+	paramSet.Modified = time.Now()
+	err = p.Parameters.Save(paramSet)
+	if err != nil {
+		return errors.Wrap(err, "unable to save parameters")
+	}
+
+	return nil
+}
+
+// ShowParameter shows the parameter set corresponding to the provided name, using
+// the provided printer.PrintOptions for display.
+func (p *Porter) ShowParameter(opts ParameterShowOptions) error {
+	paramSet, err := p.Parameters.Read(opts.Name)
+	if err != nil {
+		return err
+	}
+
+	switch opts.Format {
+	case printer.FormatJson:
+		return printer.PrintJson(p.Out, paramSet)
+	case printer.FormatYaml:
+		return printer.PrintYaml(p.Out, paramSet)
+	case printer.FormatTable:
+		// Set up human friendly time formatter
+		now := time.Now()
+		tp := dtprinter.DateTimePrinter{
+			Now: func() time.Time { return now },
+		}
+
+		// Here we use an instance of olekukonko/tablewriter as our table,
+		// rather than using the printer pkg variant, as we wish to decorate
+		// the table a bit differently from the default
+		var rows [][]string
+
+		// Iterate through all ParameterStrategies and add to rows
+		for _, pset := range paramSet.Parameters {
+			sourceVal, sourceType := GetParameterSourceValueAndType(pset.Source)
+			rows = append(rows, []string{pset.Name, sourceVal, sourceType})
+		}
+
+		// Build and configure our tablewriter
+		table := tablewriter.NewWriter(p.Out)
```

Comment:
Can we make a reusable function to initialize this table here any in any other places that use this?

Reply:
Good call; created a follow-up PR from this comment.